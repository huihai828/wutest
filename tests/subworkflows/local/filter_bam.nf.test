nextflow_workflow {

    name "Test Workflow FILTER_BAM"
    script "subworkflows/local/filter_bam.nf"
    workflow "FILTER_BAM"

    test("Should run without failures") {

        when {
            params {
                // define parameters here
                outdir = "$outputDir"
            }
            workflow {
                """
                // define inputs of the workflow here
                input[0] = [
                            [ id: 'test'],
                            [ file("assets/data/sorted_mt.bam")],
                            [ file("assets/data/sorted_mt.bam.bai")]
                        ]
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }
}