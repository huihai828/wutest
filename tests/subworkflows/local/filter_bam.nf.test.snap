{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.bam:md5,bdb7e02a8b3957560ab55f507663388b"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.bam.bai:md5,fa8edbb98f4fa2f74da0692a399181b1"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    "versions.yml:md5,d2dcfceedca9886048645a0305f030b0",
                    "versions.yml:md5,9007dbec5d82177b84b64777d8246a56"
                ],
                "bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.bam.bai:md5,fa8edbb98f4fa2f74da0692a399181b1"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.filtered.bam:md5,bdb7e02a8b3957560ab55f507663388b"
                    ]
                ],
                "csi": [
                    
                ],
                "versions": [
                    "versions.yml:md5,d2dcfceedca9886048645a0305f030b0",
                    "versions.yml:md5,9007dbec5d82177b84b64777d8246a56"
                ]
            }
        ],
        "timestamp": "2023-11-21T12:07:35+0000"
    }
}